File: cmd/gomander/frontend/src/testing/builders/command.ts
import type { Command } from "@/contracts/types.ts";

export class CommandBuilder {
	private data: Command = {
		id: crypto.randomUUID(),
		projectId: crypto.randomUUID(),
		name: "Test Command",
		command: "echo 'hello'",
		workingDirectory: "/app",
		position: 0,
		link: "",
		errorPatterns: [],
	};

	withId(id: string): this {
		this.data.id = id;
		return this;
	}

	withProjectId(projectId: string): this {
		this.data.projectId = projectId;
		return this;
	}

	withName(name: string): this {
		this.data.name = name;
		return this;
	}

	withCommand(command: string): this {
		this.data.command = command;
		return this;
	}

	withWorkingDirectory(workingDirectory: string): this {
		this.data.workingDirectory = workingDirectory;
		return this;
	}

	withPosition(position: number): this {
		this.data.position = position;
		return this;
	}

	withLink(link: string): this {
		this.data.link = link;
		return this;
	}

	withErrorPatterns(...errorPatterns: string[]): this {
		this.data.errorPatterns = errorPatterns;
		return this;
	}

	build(): Command {
		return { ...this.data, errorPatterns: [...this.data.errorPatterns] };
	}
}

File: cmd/gomander/transport/domain/command.go
package domain

import (
	commanddomain "gomander/internal/command/domain"
)

type Command struct {
	Id               string   `json:"id"`
	ProjectId        string   `json:"projectId"`
	Name             string   `json:"name"`
	Command          string   `json:"command"`
	WorkingDirectory string   `json:"workingDirectory"`
	Position         int      `json:"position"`
	Link             string   `json:"link"`
	ErrorPatterns    []string `json:"errorPatterns"`
}

func FromCommand(command commanddomain.Command) Command {
	return Command{
		Id:               command.Id,
		ProjectId:        command.ProjectId,
		Name:             command.Name,
		Command:          command.Command,
		WorkingDirectory: command.WorkingDirectory,
		Position:         command.Position,
		Link:             command.Link,
		ErrorPatterns:    command.ErrorPatterns,
	}
}

func FromCommands(commands []commanddomain.Command) []Command {
	return mapSlice(commands, FromCommand)
}

func (c Command) ToDomain() commanddomain.Command {
	return commanddomain.Command{
		Id:               c.Id,
		ProjectId:        c.ProjectId,
		Name:             c.Name,
		Command:          c.Command,
		WorkingDirectory: c.WorkingDirectory,
		Position:         c.Position,
		Link:             c.Link,
		ErrorPatterns:    c.ErrorPatterns,
	}
}
